File: src/main/java/net/gamershub/kitpvp/companions/impl/ZombifiedPiglinCompanion.java
package net.gamershub.kitpvp.companions.impl;

import net.gamershub.kitpvp.companions.Companion;
import net.minecraft.world.entity.EntityType;
import net.minecraft.world.entity.PathfinderMob;
import net.minecraft.world.entity.monster.ZombifiedPiglin;
import net.minecraft.world.level.Level;

public class ZombifiedPiglinCompanion extends Companion {
    public ZombifiedPiglinCompanion() {
        super("Zombified Piglin", 5);
    }

    @Override
    public PathfinderMob createVanillaMob(Level world) {
        ZombifiedPiglin companion = new ZombifiedPiglin(EntityType.ZOMBIFIED_PIGLIN, world);
        companion.setBaby(true);
        return companion;
    }
}

File: src/main/kotlin/io/github/chaosdave34/kitpvp/companions/impl/ZombifiedPiglinCompanion.kt
package io.github.chaosdave34.kitpvp.companions.impl

import io.github.chaosdave34.kitpvp.companions.Companion
import net.minecraft.world.entity.EntityType
import net.minecraft.world.entity.PathfinderMob
import net.minecraft.world.entity.monster.ZombifiedPiglin
import net.minecraft.world.level.Level

class ZombifiedPiglinCompanion : Companion("Zombified Piglin", 5.0) {
    override fun createVanillaMob(world: Level): PathfinderMob {
        val companion = ZombifiedPiglin(EntityType.ZOMBIFIED_PIGLIN, world)
        companion.isBaby = true
        return companion
    }
}